fvm/bootstrap.go

```diff
@@ -410,14 +447,19 @@ func (b *bootstrapExecutor) Execute() error {
 	// deploy staking collection contract to the service account
 	b.deployStakingCollection(service, &env)
 
+	// sets up the EVM environment
+	b.setupEVM(service, fungibleToken, flowToken)
+
+	err = expectAccounts(systemcontracts.EVMStorageAccountIndex)
+	if err != nil {
+		return err
+	}
+
 	b.registerNodes(service, fungibleToken, flowToken)
 
 	// set the list of nodes which are allowed to stake in this network
 	b.setStakingAllowlist(service, b.identities.NodeIDs())
 
-	// sets up the EVM environment
-	b.setupEVM(service, fungibleToken, flowToken)
```

Comment:
Ah, so this is why the indices were off last time. We set up the EVM after creating the node accounts 💡

Reply:
Yes! I did not see that registerNodes creates new accounts.

Reply:
Maybe leave a warning in the code to make it more obvious that `registerNodes` creates accounts (maybe even improve naming) and maybe that contracts should be deployed before it.